### ISSUE TITLE
Tools for parameter sensitivity

### ISSUE THREAD

**Issue by BaohaoLiao** [fe1ixxu/Intra-Distillation#1]
Hi,

this work is impressive. May I ask which tool you use for distinguishing the high and low sensitive parameters?

**Comment by fe1ixxu** [comment]
Thanks for the interest! We don't use any tools. We derive sensitivity and sort them by simply multiplying the gradient of the parameter and its magnitude (as described in Eq 2). One can implement it after the backward of the optimizer, e.g., after this line in the MT task: https://github.com/fe1ixxu/Intra-Distillation/blob/master/Machine_Translation/fairseq/tasks/intra_distillation.py#L206

**Comment by BaohaoLiao** [comment]
I'm still confused about the sensitive parameters. Are the top 20% most sensitive parameters fixed for the model? 

From Figure 2, you draw the mean sensitivity of the top 20% most sensitive parameters along the number of updates. I suppose for each update step, you will use Equation 2 to select the top 20% sensitive parameters and compute the mean sensitivity. In this way, the top 20% most sensitive parameters are not fixed. Do I understand correctly?

**Comment by fe1ixxu** [comment]
You understand correctly. They are not fixed. The purpose of Figure 2 is trying to show the contribution gap between the high- and low-sensitive parameters along the the training steps.
